tarodb/sqlite/migrations/000002_assets.up.sql

```diff
@@ -172,7 +172,7 @@ CREATE TABLE IF NOT EXISTS assets (
 
     anchor_utxo_id INTEGER REFERENCES managed_utxos(utxo_id),
     
-    UNIQUE(genesis_id, script_key_id)
+    UNIQUE(asset_id, genesis_id, script_key_id)
```

Comment:
👍

Reply:
> LGTM parrot
> 
> Nice work! Only lingering comment is if we actually want to explicitly allow a zero value split leaf (seems like it can make things non-canonical as you insert an extra leaf in there that doesn't affect the root sum amount), or if we want to change the call site, or even assert that split != root to simplify the logic there and spec impact.

Yes, that was the call site I was referencing before; IIUC, you're proposing not having `validateSplit()` called on the root split leaf in the split commitment?

I think the problem in that case is that we would then always not verify the root split leaf in the split commitment, even in cases where it was non-zero value. I agree we could do that without this extra condition here though - maybe a separate `validateSplitRoot()` or so.

I agree, no strong reason to explicitly allow non-root zero-value split leaves with this right now. FWIW, non-zero non-root split leaves is only enforced when making a split rn, not in the VM, IIRC.